dds/DCPS/MultiTask.h

```diff
@@ -0,0 +1,166 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_MULTI_TASK_H
+#define OPENDDS_DCPS_MULTI_TASK_H
+
+#include "RcEventHandler.h"
+#include "ReactorInterceptor.h"
+#include "TimeTypes.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class MultiTask : public RcEventHandler {
+public:
+  explicit MultiTask(RcHandle<ReactorInterceptor> interceptor, const TimeDuration& period)
+    : interceptor_(interceptor)
+    , period_(period)
+    , timer_(-1)
+    , next_time_()
+    , cancel_estimate_()
+  {
+    reactor(interceptor->reactor());
+  }
+
+  virtual ~MultiTask() {}
+
+  void enable(const TimeDuration& period)
+  {
+    interceptor_->execute_or_enqueue(new ScheduleEnableCommand(this, period));
+  }
+
+  void disable()
+  {
+    interceptor_->execute_or_enqueue(new ScheduleDisableCommand(this));
+  }
+
+  void disable_and_wait()
+  {
+    ReactorInterceptor::CommandPtr command = interceptor_->execute_or_enqueue(new ScheduleDisableCommand(this));
+    command->wait();
+  }
+
+  virtual void execute(const MonotonicTimePoint& now) = 0;
+
+private:
+  RcHandle<ReactorInterceptor> interceptor_;
+  const TimeDuration period_;
+  long timer_;
+  MonotonicTimePoint next_time_;
+  TimeDuration cancel_estimate_;
+  mutable ACE_Thread_Mutex mutex_;
+
+  struct ScheduleEnableCommand : public ReactorInterceptor::Command {
+    ScheduleEnableCommand(MultiTask* multi_task, const TimeDuration& period)
```

Comment:
I believe period here should properly be named delay an so on throughout the code.